### PR TITLE
fprettify: init at 0.3.7

### PR DESCRIPTION
**Pull request by FabianGD**
###### Motivation for this change

This PR adds the fprettify Formatter for Fortran sources, which is a very helpful tool when working with Fortran.

(I also add myself as the maintainer of the package, is it ok to do this in the same PR?)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/fprettify/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, python3Packages, fetchFromGitHub }:
```
Comment (figsoda) [inline pkgs/development/tools/fprettify/default.nix:1]:
```suggestion
{ lib, python3Packages, fetchFromGitHub, stdenvNoCC }:
```

File: pkgs/development/tools/fprettify/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        sha256 = "17v52rylmsy3m3j5fcb972flazykz2rvczqfh8mxvikvd6454zyj";
+    };
+
+    propagatedBuildInputs = with python3Packages; [
+        configargparse
+    ];
+
+    # The checks use subprocess.py, which fails with a FileNotFoundError
+    # on Linux. On macOS, it works just fine.
+    doCheck = false;
```
Comment (figsoda) [inline pkgs/development/tools/fprettify/default.nix:20]:
```suggestion
    doCheck = stdenvNoCC.isDarwin;
```

**Review by figsoda — CHANGES_REQUESTED** [review]
Squash commits to
`maintainers: add fabiangd`
`fprettify: init at 0.3.7`

**Comment by figsoda** [comment]
This fixes tests on linux for me
```nix
  preConfigure = ''
    patchShebangs fprettify.py
  '';
```

**Comment by figsoda** [comment]
Also, use 2 spaces for indentation

**Comment by FabianGD** [comment]
> This fixes tests on linux for me
> 
> ```nix
>   preConfigure = ''
>     patchShebangs fprettify.py
>   '';
> ```

I tried that: Great, it also fixes it for me.

File: pkgs/development/tools/fprettify/default.nix (hunk 2 above)
Comment (FabianGD) [reply pkgs/development/tools/fprettify/default.nix]: This is not necessary anymore, thanks to your fix. :)

File: pkgs/development/tools/fprettify/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  preConfigure = ''
+    patchShebangs fprettify.py
+  '';
+
+  propagatedBuildInputs = with python3Packages; [
+    configargparse
+  ];
+
+  doCheck = true;
```
Comment (figsoda) [inline pkgs/development/tools/fprettify/default.nix:22]:
```suggestion
```

this is the default

**Review by figsoda — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134423` at 05bab87c run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>fprettify</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `stdenvNoCC`.
Near pkgs/development/tools/fprettify/default.nix:1:8:
```
  |
1 | { lib, stdenvNoCC, python3Packages, fetchFromGitHub }:
  |        ^
```
</li>
  </ul>
</details>

**Comment by FabianGD** [comment]
I added your requested changes, including those by @r-rmcgibbo. Furthermore, I changed the location of fprettify in the `all-packages.nix` to a more sensible place (in the same section as I found `black`).

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 134423` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fprettify</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**